### ISSUE TITLE
Type hints fail with ResourceLoader

### ISSUE THREAD

**Issue by SirRamEsq** [godotengine/godot#33598]
**Godot version:**
3.1.1

**OS/device including version:**
Linux Mint 16

**Issue description:**
I've been writing a small library for functional programming.
One issue I encounted is that, while the ResourceLoader DOES inherit from Object ([At least according to the docs](https://docs.godotengine.org/en/3.1/classes/class_resourceloader.html)) the type hint system appears to be unaware of this.
I discovered this while writing a wrapper around ResourceLoader.load

**Steps to reproduce:**
Attempt to run the following code:
```
func loader(thing:Object, path:String):
	return thing.load(path)

func test_resource_loader_type_hint():
	yield(get_tree(), "physics_frame")

	var path = "path"
	var rsc = ResourceLoader.load(path)
	print(rsc)
	if ResourceLoader is Object:
		print ("Is Object")
	else:
		print ("Is not")
	loader(ResourceLoader, path)
```
The compiler will complain that the line `loader(ResourceLoader, path)` cannot be run because:
```
Parse Error: At 'loader()' call, argument 1. Assigned type (ResourceLoader) doesn't match the function argument's type (Object).
 modules/gdscript/gdscript.cpp:580 - Method/Function Failed, returning: ERR_PARSE_ERROR
```

If that line is commented out and the code is run, **"Is Object" Will be printed**

If any more information is required, let me know and I'll make sure to post it

**Minimal reproduction project:**
[TypeHintResourceLoaderError.zip](https://github.com/godotengine/godot/files/3841768/TypeHintResourceLoaderError.zip)

**Comment by bojidar-bg** [comment]
Duplicate of #32302?

**Comment by KoBeWi** [comment]
^
Indeed, fixed in 3.2
```
[Object:null]
Is Object
```
